pkgs/flox-activation-scripts/default.nix

```diff
@@ -102,18 +93,47 @@ runCommandNoCC "flox-activation-scripts"
       $out/activate.d/generate-fish-startup-commands.bash \
       $out/activate.d/generate-tcsh-startup-commands.bash \
       $out/activate.d/set-prompt.bash \
-      $out/etc/profile.d/* \
-      $build_wrapper/activate \
-      $build_wrapper/activate.d/generate-bash-startup-commands.bash \
-      $build_wrapper/activate.d/generate-fish-startup-commands.bash \
-      $build_wrapper/activate.d/generate-tcsh-startup-commands.bash \
-      $build_wrapper/activate.d/set-prompt.bash \
-      $build_wrapper/etc/profile.d/*
+      $out/activate.d/source-profile-d.bash \
+      $out/etc/profile.d/*
 
     # Finally check the formatting of the scripts with shfmt. Only check
     # $out as the contents of $build_wrapper will be virtually identical.
     cp ${editorconfig} $out/.editorconfig
     # This will only catch extensions and shebangs that `shfmt --find` knows about.
     ${shfmt}/bin/shfmt --diff $out
     rm $out/.editorconfig
+
+    # Remove the wrapper script from the output,
+    # we dont need that in environment interpreters.
+    rm $out/wrapper
+
+    # Next create the (lesser) "build_wrapper" output.
+    # TODO: come up with neater way to master activate script for build_wrapper case.
+
+    mkdir -p $build_wrapper
+    cp ${activation-scripts}/wrapper $build_wrapper/
+
+    # create activate.d directory and copy the required scripts
+    mkdir -p $build_wrapper/activate.d
+    cp ${activation-scripts}/activate.d/source-profile-d.bash $build_wrapper/activate.d/
+    cp ${activation-scripts}/activate.d/trace.bash $build_wrapper/activate.d/
+
+    # copy the etc/profile.d directory
+    cp -R ${activation-scripts}/etc $build_wrapper/
+
+    # make the wrapper and trace script executable
+    chmod +x $build_wrapper/wrapper
+    patchShebangs $build_wrapper/wrapper
+
+    mv $build_wrapper/activate.d/trace.bash $build_wrapper/activate.d/trace
+    chmod +x $build_wrapper/activate.d/trace
+    patchShebangs $build_wrapper/activate.d/trace
```

Comment:
I think explicity picking the required scripts and duplicating the chmods, is already better than copying from `$out` and removing specific files.
I don't like that we still have duplication.

Reply:
I do think it's a bit confusing to pick stuff out at build time. If I just look at the directory structure I'll be surprised/confused. I think it would be more clear if we had _activate_d_common, _activate_d_activate <- needs better name, and _activate_d_wrapper, and then we merged those at build time